bugsnag-android-core/src/main/java/com/bugsnag/android/Session.java

```diff
@@ -61,25 +63,28 @@ static Session copySession(Session session) {
     }
 
     Session(String id, Date startedAt, User user, boolean autoCaptured,
-            Notifier notifier, Logger logger) {
-        this(null, notifier, logger);
+            Notifier notifier, Logger logger, String apiKey) {
+        this(null, notifier, logger, apiKey);
         this.id = id;
         this.startedAt = new Date(startedAt.getTime());
         this.user = user;
         this.autoCaptured.set(autoCaptured);
+        this.apiKey = apiKey;
     }
 
     Session(String id, Date startedAt, User user, int unhandledCount, int handledCount,
-            Notifier notifier, Logger logger) {
-        this(id, startedAt, user, false, notifier, logger);
+            Notifier notifier, Logger logger, String apiKey) {
+        this(id, startedAt, user, false, notifier, logger, apiKey);
         this.unhandledCount.set(unhandledCount);
         this.handledCount.set(handledCount);
         this.tracked.set(true);
+        this.apiKey = apiKey;
     }
 
-    Session(File file, Notifier notifier, Logger logger) {
+    Session(File file, Notifier notifier, Logger logger, String apiKey) {
         this.file = file;
         this.logger = logger;
+        this.apiKey = apiKey;
```

Comment:
This should be extracting the `apiKey` from the filename by preference, falling back to the `apiKey` when it's not available (`file` may also be `null` here, so it's important to take that into account as well).